### PR TITLE
Support for CentOS 7 / RHEL 7

### PR DESCRIPTION
**Pull request by scollazo**
This allows for atom installs on rpm based distributions. 

- Uses rh-php70 packages from software collections repository ( nginx, rh-php70...)
- Uses rpmfree repo for ffmpeg package
- Memcached php module is built from [websupport-sk/pecl-memcache@NON_BLOCKING_IO_php7](https://github.com/websupport-sk/pecl-memcache/tree/NON_BLOCKING_IO_php7)

### CODE REVIEW

File: tasks/search.yml [hunk 1]
```diff
@@ -5,4 +5,6 @@
   shell: "php symfony search:populate 1>/dev/null"
   args:
     chdir: "{{ atom_path }}"
-  ignore_errors: "yes" # segfault in php7, not sure yet why!
+  environment:
```
Comment (hakamine) [inline tasks/search.yml]: does it completely build the index even though it segfaults? Maybe need to clarify in the comment

File: tasks/flush.yml [hunk 2]
```diff
@@ -4,11 +4,15 @@
   command: "php symfony tools:purge --no-confirmation --email='{{ atom_user_email }}' --username='{{ atom_user_username }}' --password='{{ atom_user_password }}' --title='{{ atom_title }}' --description='{{ atom_description }}' --url='http://undefined'"
   args:
     chdir: "{{ atom_path }}"
+  environment:
+    PATH: "{{ ansible_env.PATH }}:/opt/rh/rh-php70/root/bin/"
```
Comment (hakamine) [inline tasks/flush.yml]: I guess adding this to the path doesn't affect when deploying on ubuntu, but is there a way to make this addition conditional (add to path only when deploying on CentOS/RH)?. Or just add a comment to clarify.

Comment (DavidHume-ArtefactualSystems) [inline tasks/flush.yml]: I had the same thought/question here as hakamine (also for handlers/main.yml)

File: handlers/main.yml [hunk 3]
```diff
@@ -40,12 +48,20 @@
...
+
 - name: "Clear sf_cache"
   shell: "php symfony cc"
   args:
     chdir: "{{ atom_path }}"
   become_user: "{{ atom_user }}"
-  environment: "{{ atom_pool_php_envs }}"
+  environment:
+    - "{{ atom_pool_php_envs }}"
+    - PATH: "{{ ansible_env.PATH }}:/opt/rh/rh-php70/root/bin/"
```
Comment (DavidHume-ArtefactualSystems) [inline handlers/main.yml]: Similar to tasks/flush.yml, does this PATH change (/opt/rh/rh-php70/root/bin) even occur if not CentOS/RedHat? Possibly harmless if so, or should there be a conditional?

**Comment by mamedin** [comment]
I tested this role in a new deployment. I had problems flushing database, but it was fixed rebasing this branch into master. I think it needs the new commits:

- https://github.com/artefactual-labs/ansible-atom/commit/245ecf3f1c823c39017e32c6e42f47dc23d6a5bc
- https://github.com/artefactual-labs/ansible-atom/commit/c119a722c90bc5fdede3a2bc2723c9c0ef9cdcbd

Considering that, the role works perfectly in CentOS. Good job.

**Comment by hakamine** [comment]
I had been working on fixes for this branch, adding to @scollazo great initial work, but had to left the work unfinished because of other priorities. I think there were other reasons this wasn't ready to merge yet (can't recall right now). I'll try to retake this work soon (or at least try to document what was missing)

**Comment by scollazo** [comment]
Rebased against master cb402b3e20299c5b

**Comment by scollazo** [comment]
@hakamine @mamedin , this branch is being used for production deployments, and I would like to merge it before my work for 2.5. Is ok if I merge this as is, and we improve on it later?

File: tasks/deps-rh.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+- name: "Add elasticsearch repo"
+  yum_repository:
+    name: "elasticsearch-1.7"
+    description: "Elasticsearch repository for 1.7.x packages"
+    file: elasticsearch_repo
+    gpgkey: "http://packages.elastic.co/GPG-KEY-elasticsearch"
+    gpgcheck: yes
+    baseurl: "http://packages.elastic.co/elasticsearch/1.7/centos"
+
```
[the remark this answers is not in the document]
Comment (scollazo) [reply tasks/deps-rh.yml]: Yes, we are using the common elasticsearch role now

File: tasks/deps-rh.yml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+- name: "Install services "
+  yum:
+    name: "{{ item }}"
+    state: "installed"
+  with_items:
+    - java-1.8.0-openjdk
+    - gearmand
+    - elasticsearch
+    - memcached
+
```
[the remark this answers is not in the document]
Comment (scollazo) [reply tasks/deps-rh.yml]: Yes, we are using the memcached role now

**Comment by mamedin** [comment]
> @hakamine @mamedin , this branch is being used for production deployments, and I would like to merge it before my work for 2.5. Is ok if I merge this as is, and we improve on it late

Perfecto :+1:

**Comment by hakamine** [comment]
> @hakamine @mamedin , this branch is being used for production deployments, and I would like to merge it before my work for 2.5. Is ok if I merge this as is, and we improve on it later?

Ok

**Pull request merged**